src/MyMarketManager.WebApp/Program.cs

```diff
@@ -24,17 +25,35 @@
 builder.Services.AddScoped<DbContextMigrator>();
 builder.Services.AddHostedService<DatabaseMigrationService>();
 
+// Add Azure Computer Vision embedding generators as keyed services
+// Configuration comes from Aspire-provisioned resources or appsettings
+var computerVisionEndpoint = builder.Configuration.GetConnectionString("ai-foundry") ?? builder.Configuration["AzureAI:Endpoint"] ?? "";
+var computerVisionApiKey = builder.Configuration["AzureAI:ApiKey"] ?? "";
+if (!string.IsNullOrEmpty(computerVisionEndpoint) && !string.IsNullOrEmpty(computerVisionApiKey))
+{
+    builder.Services.AddAzureComputerVisionEmbeddings(computerVisionEndpoint, computerVisionApiKey);
+}
+
 // Add scraper services
 builder.Services.Configure<ScraperConfiguration>(builder.Configuration.GetSection("Scraper"));
 builder.Services.AddScoped<IWebScraperSessionFactory, WebScraperSessionFactory>();
 builder.Services.AddScoped<SheinWebScraper>();
 
-builder.Services.AddBackgroundProcessing(builder.Configuration.GetSection("BackgroundProcessing"))
+var processingBuilder = builder.Services.AddBackgroundProcessing(builder.Configuration.GetSection("BackgroundProcessing"))
     .AddHandler<SheinBatchHandler>(
         name: "Shein",
         maxItemsPerCycle: 5,
         purpose: WorkItemHandlerPurpose.Ingestion);
 
+// Only register image vectorization handler if Azure AI is configured
+if (!string.IsNullOrEmpty(computerVisionEndpoint) && !string.IsNullOrEmpty(computerVisionApiKey))
```

Comment:
We always add the service, otherwise things will throw at runtime with missing services.